tools/otci/otci/otci.py

```diff
@@ -175,30 +175,41 @@ def factory_reset(self):
     # Network Operations
     #
 
-    def ping(self, ip: str, size: int = None, count: int = None, interval: int = None, hoplimit: int = None):
-        """Send an ICMPv6 Echo Request.
+    def ping(self, ip: str, size: int = 8, count: int = 1, interval: float = 1, hoplimit: int = 64, timeout: float = 3):
+        """Send an ICMPv6 Echo Request. The default arguments are consistent with those in src/core/config/ping_sender.h.
 
         :param ip: The target IPv6 address to ping.
-        :param size: The number of data bytes in the payload.
-        :param count: The number of ICMPv6 Echo Requests to be sent.
-        :param interval: The interval between two consecutive ICMPv6 Echo Requests in seconds. The value may have fractional form, for example 0.5.
-        :param hoplimit: The hoplimit of ICMPv6 Echo Request to be sent.
+        :param size: The number of data bytes in the payload. Default is 8.
+        :param count: The number of ICMPv6 Echo Requests to be sent. Default is 1.
+        :param interval: The interval between two consecutive ICMPv6 Echo Requests in seconds. The value may have fractional form, for example 0.5. Default is 1.
+        :param hoplimit: The hoplimit of ICMPv6 Echo Request to be sent. Default is 64. See OPENTHREAD_CONFIG_IP6_HOP_LIMIT_DEFAULT in src/core/config/ip6.h.
+        :param timeout: The maximum duration in seconds for the ping command to wait after the final echo request is sent. Default is 3.
         """
-        cmd = f'ping {ip}'
-
-        if size is not None:
-            cmd += f' {size}'
-
-        if count is not None:
-            cmd += f' {count}'
-
-        if interval is not None:
-            cmd += f' {interval}'
-
-        if hoplimit is not None:
-            cmd += f' {hoplimit}'
-
-        self.execute_command(cmd)
+        cmd = f'ping {ip} {size} {count} {interval} {hoplimit} {timeout}'
+
+        timeout_allowance = 3
+        lines = self.execute_command(cmd, timeout=(count - 1) * interval + timeout + timeout_allowance)
+        packet_count_pattern = re.compile(r'(\d+) packets transmitted, (\d+) packets received.')
+        packet_loss_pattern = re.compile(r'Packet loss = (\d+\.\d+)%')
+        round_trip_time = re.compile(r'Round-trip min/avg/max = (\d+)/(\d+\.\d+)/(\d+) ms')
```

Comment:
Are these statistics separated in multiple lines?

Reply:
No. They are in the same line.

Reply:
If so we may use one line of regex.

Reply:
My concern is, the regex then may be too complex. I can try merging them though.

Reply:
Updated.